### PR TITLE
Upgrade Guide

### PR DESCRIPTION
**Pull request by oblm**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

#### Description
* Adds an upgrade guide to the public documentation.
* Adds that upgrade guide to the table of contents on the Troubleshooting section.

#### Tests
Rendered in improbadoc.

Capture

### CODE REVIEW

**Comment by ernopp** [comment]
Thanks ollie

File: SpatialGDK/Documentation/content/upgrading.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<%(TOC)%>
+# Keep your GDK up to date
+
+To use the SpatialOS GDK for Unreal, you need two git repositories:<br>
+
+* [The SpatialOS Unreal Engine fork.](https://github.com/improbableio/UnrealEngine)
+* [The GDK](https://github.com/spatialos/UnrealGDK)<br>
+
+To benefit from the latest version of GDK, always develop your game on the latest version. Whenever you update your GDK, you **must** also update your SpatialOS Unreal Engine fork. If you don't, errors can be caused by the two repositories being out of sync.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
It is good practice to always develop your game on the latest version of the software by updating the software. This ensures you benefit from the most up-to-date functionality. Whenever you update your GDK software, you **must** also update your SpatialOS Unreal Engine fork software. If you don't, you may get errors from them being out of synch.
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<%(TOC)%>
+# Keep your GDK up to date
+
+To use the SpatialOS GDK for Unreal, you need two git repositories:<br>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
To use the SpatialOS GDK for Unreal, you need software from two git repositories:<br>
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+* [The SpatialOS Unreal Engine fork.](https://github.com/improbableio/UnrealEngine)
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
* [The SpatialOS Unreal Engine fork](https://github.com/improbableio/UnrealEngine)
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+* [The GDK](https://github.com/spatialos/UnrealGDK)<br>
+
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
You download bith od these as part of the [Get started]({{urlRoot}}/content/get-started/introduction) steps. <br/>
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+To benefit from the latest version of GDK, always develop your game on the latest version. Whenever you update your GDK, you **must** also update your SpatialOS Unreal Engine fork. If you don't, errors can be caused by the two repositories being out of sync.
+
+We recommend that you update your version of the GDK and SpatialOS Unreal Engine fork every week.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
We recommend that you update your version of the GDK and SpatialOS Unreal Engine fork every week.  To do this, follow the steps below.
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+## Step 1: Ensure you're on the release branches.
+
+If you followed our [get started]({{urlRoot}}/content/get-started/introduction) guide, you have these repositories cloned on your computer.<br>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
If you followed our [Get started]({{urlRoot}}/content/get-started/introduction) guide, you have these repositories cloned on your computer.<br>
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+* Your `UnrealGDK` repository should have the `release` branch checked out.<br>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
* Your `UnrealEngine` repository should have the branch ending with `-SpatialOSUnrealGDK-release`checked out.<br>
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+* Your `UnrealEngine` repository should have the branch ending with `-SpatialOSUnrealGDK-release`checked out.<br>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
* Your `UnrealGDK` repository should have the `release` branch checked out.<br>
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+You can determine which branch is checked out by following the instructions below:<br>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
You can find out which branch you have checked out by following the instructions below:<br>
```

File: SpatialGDK/Documentation/toc.md [hunk 10]
```diff
@@ -33,6 +33,7 @@
     - [Cloud development workflow]({{urlRoot}}/content/cloud-dev-workflow)
     - [Troubleshooting]({{urlRoot}}/content/troubleshooting)
     - [Known issues]({{urlRoot}}/known-issues)
+    - [Keeping your GDK up to date]({{urlRoot}}/content/upgrading)
```
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]:
```suggestion
    - [Keep your GDK up to date]({{urlRoot}}/content/upgrading)
```

File: SpatialGDK/Documentation/content/upgrading.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
+<%(TOC)%>
+# Keep your GDK up to date
+
+To use the SpatialOS GDK for Unreal, you need software from two git repositories:<br>
+
+* [The SpatialOS Unreal Engine fork](https://github.com/improbableio/UnrealEngine)
+* [The GDK](https://github.com/spatialos/UnrealGDK)<br>
+You download both of these as part of the [Get started]({{urlRoot}}/content/get-started/introduction) steps. <br/>
+It is good practice to always develop your game on the latest version of the software by regularly updating it. This ensures you benefit from the most up-to-date functionality. Whenever you update your GDK software, you **must** also update your SpatialOS Unreal Engine fork software. If you don't, you may get errors from them being out of synch.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/upgrading.md]:
```suggestion
To ensure you benefit from the most up-to-date functionality, always develop your game on the latest version of the software by regularly updating it. Whenever you update your GDK software, you **must** also update your SpatialOS Unreal Engine fork software. If you don't, you might get errors from them being out of synch.
```

**Pull request merged**